### PR TITLE
Enhance Filtering Documentation

### PR DESCRIPTION
**Pull request by Jitsusama**
This PR is related to issue #56.

It seeks to significantly expand upon the `filters` parameter documentation in order to hopefully make understanding this property much easier for the users of this excellent plugin.

### CODE REVIEW

**Comment by Jitsusama** [comment]
I think it's safe to say that the build failure wasn't due to the markdown changes I made. It looks like your linter is not very happy.

**Comment by errm** [comment]
Yeah, don't worry about the linter ... it is fixed on the 1.0 branch... we don't care here though ...

File: docs/Filtering-Details.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Filtering Details
+
+## Overview
+
+This application takes an array of hashes passed to the `filters` parameter
+within a `systemd` typed source definition in your `fluent.conf` configuration
+file and then parses them into a format understood by `systemd`'s `journald`
```
Comment (errm) [inline docs/Filtering-Details.md]: I am not sure that it is strictly true to describe this as the `journald` API...

I think "libsystemd's journal API" would be slightly more accurate description...

**Comment by Jitsusama** [comment]
Groovy. Let me know if you have any questions or corrections or even state how much you despise these changes :smile:.

**Review by errm — COMMENTED** [review]
This seems like a great start ...

I am a little worried that terms are getting a little overloaded here...

The filters param that we are discussing here is the native filtering facility provided by libsystemd so the plugin natively only reads the messages from the journal that we are interested in.

This plugin also includes a fluentd filter plugin that can be used to munge journal entries by modifying the structure of each entry ... 

Either we need to find a way to be super clear about what is what or perhaps we should change the name of this parameter to clear up the confusion ...

I think it would be equally self explanatory to name this param `matches` in line with the journalctl documentation.... but equally something like `query` could also make sense WDYT?

**Comment by Jitsusama** [comment]
@errm; I'm still a newb with fluentd, so I was not aware of the name overload. I agree, removing ambiguity is ideal, and sticking with domain language is always a good thing. With that said, I think sticking with `journalctl`'s _matches_ would make the most sense.

Now, changing the name would be a breaking change, so that should probably be done in the 1.0.0 branch. You might even want to continue supporting both in 1.0.0 until you release 1.1.0 or something. Would you like for me to update my V1-Enhance-Filtering-Documentation branch to refer to it as `matches` instead, and leave the master branch alone (in regards to this particular change)?

**Comment by errm** [comment]
That sounds like a plan ...

We have the facility to deprecate params so it should be simple to change the name but keep the old one working too (for now)

**Comment by Jitsusama** [comment]
I've updated this PR with what I hope are the requested changes, and also updated PR #58 with the same, while also adding details on the `filters` deprecation/replacement with `matches`.

**Comment by errm** [comment]
Great... this is a nice addition, so I will merge it straight away ... I will need to fix the implementation first on the v1 branch :)

Thanks so much for helping out ... the docs were the only thing really left to fixup here so this is super useful..

**Comment by Jitsusama** [comment]
You are very welcome. And I must say, this has been one of my more pleasant PR experiences. Thanks for being so speedy in responding and clearly defining your desires!

**Pull request merged**